modelling/src/main/java/org/axonframework/modelling/entity/child/AbstractEntityChildModel.java

```diff
@@ -142,35 +141,47 @@ protected Builder(@Nonnull Class<P> parentClass,
         }
 
         /**
-         * Sets the {@link ChildEntityMatcher} to use for matching the child entities to the command. This is used to
-         * filter the child entities based on the command. One entity must match an incoming command. If no match is
-         * found, an {@link ChildEntityNotFoundException} is thrown. If multiple matches are found, an
-         * {@link ChildAmbiguityException} is thrown.
+         * Sets the {@link CommandTargetResolver} to use for resolving the child entity to handle the command. This
+         * should return one child entity, or {@code null} if no child entity should handle the command.
+         * <p>
+         * Defaults to matching a singular child entity, or throwing a {@link ChildAmbiguityException} if more than one
+         * child candidate exists. As such, provide another implementation if there are multiple candidates, such as
+         * when the member is a {@link List} of child entities.
          *
-         * @param commandTargetMatcher The {@link ChildEntityMatcher} to use for matching the child entities to the
-         *                             command.
+         * @param commandTargetResolver The {@link CommandTargetResolver} to use for resolving the child entity
+         *                              to handle the command.
          * @return This builder instance.
          */
         @SuppressWarnings("unchecked")
-        public R commandTargetMatcher(
-                @Nonnull ChildEntityMatcher<C, CommandMessage<?>> commandTargetMatcher) {
-            this.commandTargetMatcher = requireNonNull(commandTargetMatcher,
-                                                       "The commandTargetMatcher may not be null.");
+        public R commandTargetResolver(@Nonnull CommandTargetResolver<C> commandTargetResolver) {
+            this.commandTargetResolver = requireNonNull(commandTargetResolver,
+                                                        "The commandTargetResolver may not be null.");
             return (R) this;
         }
 
+        protected void validate() {
+            if (commandTargetResolver == null) {
```

Comment:
These two if-checks would still be use cases of the old `BuilderUtils` class, as it automatically asserts and throws an `AxonConfigurationException`

Reply:
Fair enough, applying the `BuilderUtils`